src/services/expenses.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const { Expense } = require('../utils/models/Expense');
+
+const getFilteredExpenses = async(searchParams) => {
+  const {
+    userId,
+    category,
+    from,
+    to,
+  } = searchParams;
+
+  const expenses = await Expense.findAll();
+
+  return expenses
+    .filter(expense => {
+      if (userId && expense.userId !== +userId) {
+        return;
+      }
+
+      if (category && category !== expense.category) {
+        return;
+      }
+
+      if (from && new Date(expense.spentAt) < new Date(from)) {
+        return;
+      }
+
+      if (to && new Date(expense.spentAt) > new Date(to)) {
+        return;
+      }
+
+      return true;
+    });
```

Comment:
strange that u don't use SQL filters. I think u need to rewrite this one on sequelize SQL calls .findAll with clauses